api/src/routes/ranks.ts

```diff
@@ -0,0 +1,287 @@
+ *         schema:
+ *           type: integer
+ *       - in: query
+ *         name: gameId
+ *         required: true
+ *         description: ID of the game to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *       500:
+ *         description: Internal Server Error
+ */
+
+// TODO: Add a since/until parameter.
+router.get("/average", async (req: Request, res: Response) => {
+  try {
+    const { userId, gameId } = req.query;
+
+    const result = await pool.query(
+      "SELECT * FROM user_ranks WHERE user_id = $1 AND game_id = $2",
+      [userId, gameId]
+    );
+
+    const ranks = result.rows;
+    res.json(await getAverageRank(ranks));
+  } catch (error) {
+    logger.error(error);
+    res.status(500).send("Internal Server Error");
+  }
+});
+
+/**
+ * @swagger
+ * /ranks/connections:
+ *   get:
+ *     summary: Get all ranks of a user and their connections on a game
+ *     tags: [Ranks]
+ *     description: Retrieve all ranks of a user and their connections on a specific game from the database.
+ *     parameters:
+ *       - in: query
+ *         name: userId
+ *         required: true
+ *         description: ID of the user to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *       - in: query
+ *         name: gameId
+ *         required: true
+ *         description: ID of the game to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *       - in: query
+ *         name: page
+ *         description: Page number for pagination
+ *         schema:
+ *           type: integer
+ *         default: 1
+ *       - in: query
+ *         name: pageSize
+ *         description: Number of logs per page
+ *         schema:
+ *           type: integer
+ *         default: 10
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *       500:
+ *         description: Internal Server Error
+ */
+
+// TODO: Add a since/until parameter.
+router.get("/connections", async (req: Request, res: Response) => {
+  try {
+    const { userId, gameId, page = 1, pageSize = 10 } = req.query;
+
+    if (!paginationIsValid(page as number, pageSize as number)) {
+      return res
+        .status(400)
+        .json({ error: "Invalid page or pageSize parameters" });
+    }
+
+    // Calculate the offset for pagination
+    const offset = (Number(page) - 1) * Number(pageSize);
+
+    const userConnections = await pool.query(
+      "SELECT user1_id, user2_id FROM user_connections WHERE user1_id = $1 OR user2_id = $1",
+      [userId]
+    );
+    let connectionIds = userConnections.rows.map((connection) => [
+      connection.user1_id,
+      connection.user2_id,
+    ]);
+
+    connectionIds = [...new Set(connectionIds.flat())];
+
+    const result = await pool.query(
+      `SELECT * FROM user_ranks WHERE user_id IN (${connectionIds.join(
```

Comment:
Instead of embedding the .join here can you not do that when you redefine `connectionIds`, it'll make the query a lot cleaner?